modules/dhcp_native_ms/dhcp_native_ms_main.rb

```diff
@@ -74,6 +74,7 @@ def unused_ip(subnet_address, mac_address, from_address, to_address)
       client = dhcpsapi.get_client_by_mac_address(subnet_address, mac_address) rescue nil
       return client[:client_ip_address] unless client.nil?
 
+      raise NoMethodError.new("DhcpsApi::Server#get_free_ip_address is not available on Windows Server 2008 and earlier versions.") unless dhcpsapi.api_level == DhcpsApi::Server::DHCPS_WIN2012_API
```

Comment:
1. NoMethodError's not appropriate here, could you use Proxy::DHCP::Error?
1. Perhaps not necessary to change now, but if there are any future API versions then it'd be useful to either _blacklist_ 2008 and "none" API levels (rather than whitelist 2012), or have the API level comparable so it checks for versions < 2012.